### PR TITLE
feat(docs): use algolia docsearch

### LINKED ISSUE #1720: [search] use algolia docsearch
## 📚 Documentation
While built-in search is enough, searching something and hit enter feels like not a good UX.
To be able to search interactively, we can apply [Algolia Docsearch](https://docsearch.algolia.com/) for that.
PyTorch is also using it when searching from homepage.

### PR DESCRIPTION
**Pull request by ydcjeff**
Fixes #1720
https://deploy-preview-1727--pytorch-ignite-preview.netlify.app
Description: Add algolia docsearch for interactive searching

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by ydcjeff** [comment]
From ignite community, feel free to visit, try searching a bit and comment any problem you met. This PR is in WIP.
Current problem I found
- [ ] search UI small
- [ ] showing up many versions (v0.2.0 showed up when searching from master) of search results

**Comment by vfdev-5** [comment]
Thanks for integrating it @ydcjeff !
For me UI thing is OK. And yes, I agree that we may would like to restrict the search by the current latest version or master ?

File: docs/source/conf.py [hunk 1]
```diff
@@ -32,8 +32,8 @@
     version = os.environ["code_version"]
     if "master" in version:
         version = "master (" + ignite.__version__ + ")"
-    else:
-        version = version.replace("v", "")
+    # else:
+    #     version = version.replace("v", "")
```
Comment (ydcjeff) [inline docs/source/conf.py:36]: This was commented out for algolia search to work with each version.

File: docs/source/_templates/layout.html [hunk 2]
```diff
@@ -483,5 +483,21 @@ <h2>Resources</h2>
       });
     })
   </script>
+  <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+  <script type="text/javascript">
+  let VERSION
+  if ('{{ version }}'.includes('v')) {
+    VERSION = '{{ version }}'
+  } else {
+    VERSION = '{{ release }}'
```
Comment (ydcjeff) [inline docs/source/_templates/layout.html:492]: `{{ version }}` is from `conf.py`. If a version include `v`, then it's a tag else we will use `release` which is `master` as algolia stated 
```
So as of today you have: master, v0.1.0, v0.1.1, v0.1.2, v0.2.0, v0.2.1, v0.3.0, v0.4.0.post1, v0.4.1, v0.4.2, v0.4.3, v0.4rc.0.post1
```

File: docs/source/_templates/searchbox.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+{% else %}
+  {% set search_project = 'Docs' %}
+{% endif %}
+
+<div role="search">
+  <form id="rtd-search-form" class="wy-form">
+    <input type="text" placeholder="Search {{ search_project }}" />
+  </form>
+</div>
+{%- endif %}
```
Comment (ydcjeff) [inline docs/source/_templates/searchbox.html:14]: copied from `searchbox.html` of `pytorch_sphinx_theme`, removed unnecessary lines.

**Comment by ydcjeff** [comment]
To test for searching works with each version, visit https://ydcjeff.github.io/ignite/
That was deployed with this PR on my fork

**Comment by vfdev-5** [comment]
@ydcjeff looks great now ! Thanks for the update ! Can we merge it now or I think to work more on that ?

**Comment by ydcjeff** [comment]
@vfdev-5 we can merge now
We will update it from the community feedback.

**Pull request merged**